### PR TITLE
Support multi-column ``groupby`` with ``sort=True`` and ``split_out>1``

### PR DESCRIPTION
**Pull request by rjzamora**
Uses the multi-column `sort_values` support added in https://github.com/dask/dask/pull/8263 to support multi-column groupby aggregations with `sort=True` and `split_out > 1`.

### CODE REVIEW

File: dask/dataframe/tests/test_groupby.py [hunk 1]
```diff
@@ -1230,21 +1230,20 @@ def test_shuffle_aggregate_sort(shuffle_method, sort):
...
+    )
+    expect = pdf.groupby(["a", "b"], sort=sort).agg(spec)
+    assert_eq(expect, result)
     if sort:
-        # Cannot use sort=True with multiple groupby keys
-        with pytest.raises(NotImplementedError):
-            ddf.groupby(["a", "b"], sort=sort).agg(
-                spec, split_out=2, shuffle=shuffle_method
-            )
+        assert (result.index.values.compute() == expect.index.values).all()
```
Comment (hendrikmakait) [inline dask/dataframe/tests/test_groupby.py:1246]: How does this differ from setting `assert_eq(..., sort_results=not sort)`?

**Review by hendrikmakait — COMMENTED** [review]
The code generally looks good, with one question regarding the tests. I'd like someone with more dataframe experience to review before signing off (cc @phofl).

**Review by phofl — COMMENTED** [review]
lgtm

File: dask/dataframe/tests/test_groupby.py (hunk 1 above)
Comment (rjzamora) [reply dask/dataframe/tests/test_groupby.py]: Yeah, good point. We don't really need this `if sort` block at all if we just add `sort_results=not sort` to the `assert_eq` above. Changed.

**Comment by rjzamora** [comment]
Thanks for reviewing @hendrikmakait  and @phofl !

**Pull request merged**